packages/shader-lab/src/preprocessor/MacroDefine.ts

```diff
@@ -1,6 +1,6 @@
 import { BaseToken } from "../common/BaseToken";
 import { ShaderRange } from "../common";
-import { ParserUtils } from "../Utils";
+import { GSError, GSErrorName } from "../GSError";
 
 export class MacroDefine {
```

Comment:
Use `public readonly macro: BaseToken` to opt in constructor

Reply:
done